src/main/resources/templates/kotlin-lang/kotlinClassGraphqlParametrizedInput.ftl

```diff
@@ -0,0 +1,59 @@
+<#assign MapperUtil=statics["com.kobylynskyi.graphql.codegen.kotlin.KotlinGraphQLTypeMapper"]>
+<#if package?has_content>
+package ${package}
+
+</#if>
+import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLParametrizedInput
+import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLRequestSerializer
+import java.util.StringJoiner
+<#if equalsAndHashCode>
+import java.util.Objects
+</#if>
+
+<#if javaDoc?has_content>
+/**
+<#list javaDoc as javaDocLine>
+ * ${javaDocLine}
+</#list>
+ */
+</#if>
+<#if generatedInfo.getGeneratedType()?has_content>
+@${generatedInfo.getGeneratedType()}(
+    value = ["com.kobylynskyi.graphql.codegen.GraphQLCodegen"],
+    date = "${generatedInfo.getDateTime()}"
+)
+</#if>
+<#list annotations as annotation>
+@${annotation}
+</#list>
+data class ${className}(
+<#if fields?has_content>
+<#list fields as field>
+    <#if field.deprecated>
+        @java.lang.Deprecated
+    </#if>
+    <#list field.annotations as annotation>@get:${annotation}
+    </#list>val ${field.name}: ${field.type}<#if field.defaultValue?has_content> = ${field.defaultValue}</#if><#if field_has_next>,</#if>
+</#list>
+</#if>
+) : GraphQLParametrizedInput {
+
+    override fun toString(): String {
+        val joiner = StringJoiner(", ", "(", ")")
+<#if fields?has_content>
+<#list fields as field>
+        <#if field.type?ends_with("?")>
+            <#if MapperUtil.isKotlinPrimitive(field.type)>
+        joiner.add("${field.originalName}: " + GraphQLRequestSerializer.getEntry(${field.name}))
+            <#else>
+        if (${field.name} != null) {
+            joiner.add("${field.originalName}: " + GraphQLRequestSerializer.getEntry(${field.name}))
+        }
```

Comment:
Could you please explain this part? Why would the field name be null when the field is not a primitive type?

Reply:
This is the same as Java

Reply:
Then this is wrong I believe. It makes sense in Java because primitive = non null, and non-primitive = nullable. The check to decide whether to put a null check or not in Kotlin should be entirely determine by whether the type is a nullable type or not (`?`) and should not have anything to do with primitives.

Reply:
In fact, I'm not sure if this method is used for the request api, otherwise I think we should remove the method.